File: dmn-test-cases/standard/proto/1.1/0004-lending/translator/expected/proto3/java/dmn/type/TBureauDataImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tBureauData"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TBureauDataImpl implements TBureauData {
        private java.lang.Number creditScore;
        private Boolean bankrupt;

    public TBureauDataImpl() {
    }

    public TBureauDataImpl(Boolean bankrupt, java.lang.Number creditScore) {
        this.setBankrupt(bankrupt);
        this.setCreditScore(creditScore);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Bankrupt")
    public Boolean getBankrupt() {
        return this.bankrupt;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("Bankrupt")
    public void setBankrupt(Boolean bankrupt) {
        this.bankrupt = bankrupt;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("CreditScore")
    public java.lang.Number getCreditScore() {
        return this.creditScore;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("CreditScore")
    public void setCreditScore(java.lang.Number creditScore) {
        this.creditScore = creditScore;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/standard/proto/1.1/0004-lending/translator/expected/proto3/kotlin/dmn/type/TBureauDataImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tBureauData"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TBureauDataImpl : TBureauData {
    @get:com.fasterxml.jackson.annotation.JsonGetter("CreditScore")
    @set:com.fasterxml.jackson.annotation.JsonGetter("CreditScore")
    override var creditScore: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Bankrupt")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Bankrupt")
    override var bankrupt: Boolean? = null

    constructor() {
    }

    constructor (bankrupt: Boolean?, creditScore: kotlin.Number?) {
        this.bankrupt = bankrupt
        this.creditScore = creditScore
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
